### ISSUE TITLE
Omitting `data` property from `relationships` information

### ISSUE THREAD

**Issue by hovmand** [woohoolabs/yin#50]
I'm trying to describe a relationship only by the `links` for which can be used to fetch the actual data of the related resource.

There are two resources **A** and **B**, where **A** has many **B**s. When fetching **A** the entries of **B** shouldn't be included.

Performing the following request
```HTTP
GET /api/A/1234 HTTP/1.1
Host: test.dev
Accept: application/json
```
responds with
```JSON
{
    "jsonapi": {
        "version": "1.0"
    },
    "links": {
        "self": "/api/A/1234"
    },
    "data": {
        "type": "A",
        "id": "1234",
        "links": {
            "self": "/api/A/1234"
        },
        "attributes": {
            "name": "Resource 1"
        },
        "relationships": {
            "B": {
                "links": {
                    "self": "/api/A/1234/relationships/B",
                    "related": "/api/A/1234/B"
                },
                "data": []
            }
        }
    }
}
```
where I'm trying to get a response without the `data.relationships.B.data` property (the empty array).

The relationship is described in the transformer for **A** by:
```PHP
ToManyRelationship::create()
    ->setLinks(
        Links::createWithoutBaseUri([
            'self'    => new Link('/api/A/' . $this->getId($a) . '/relationships/B'),
            'related' => new Link('/api/A/' . $this->getId($a) . '/B'),
        ])
    )
    ->omitWhenNotIncluded()
```

The [documentation](http://jsonapi.org/format/#document-resource-object-relationships) states that a relationship must contain at least one of the three properties: `links`, `data`, or `meta`. (it's valid without the `data` property).

It seems like commit 1aef0431e6b0b9866cdb628e3d37289e7c3f1409 forces the `data` property to be present. It was implemented to resolve #22. The commit basically assigns a  variable `$x = $y`, and then later compares the two `$x === $y`.

I'm not sure about what the `Transformation::$fetchedRelationship` variable should express, but comparing issue #22 with the analogy of my case, my best guess is that it should represent resource **A**, thereby ensuring that the `data` property of **A** is present at the outer scope.

**Comment by kocsismate** [comment]
First, thank you for the report, this was an important issue. I believe the fix was easy: I only added a condition which ensures that the `data` property is only present in the response if you invoked `$relationship->setData()`.

Could you please confirm that my fix works on your end too?

**Comment by hovmand** [comment]
It works. Thanks.

**Comment by kocsismate** [comment]
Cool! I've just released v1.0.1!
